### PR TITLE
feat(linter): add fixer for eslint/func_names

### PR DESCRIPTION
**Pull request by DonIsaac**

### CODE REVIEW

**Comment by DonIsaac** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4714?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4714** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4714?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4713** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4713?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4712** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4712?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4711** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4711?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4710** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4710?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @DonIsaac and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_linter/src/rules/eslint/func_names.rs [hunk 1]
```diff
@@ -422,24 +464,89 @@ impl Rule for FuncNames {
...
+    "undefined",
+    "yield",
+};
+
+fn is_valid_identifier_name(name: &str) -> bool {
+    !INVALID_NAMES.contains(name)
+        && name.chars().all(|c| match c {
+            '_' => true,
+            c => c.is_alphanumeric(),
+        })
```
Comment (mysteryven) [inline crates/oxc_linter/src/rules/eslint/func_names.rs:539]: Maybe [`is_identifier_name`](https://github.com/oxc-project/oxc/blob/bc7ea0bedb51b47acd9c144028e98025ac53c3dd/crates/oxc_syntax/src/identifier.rs#L139-L142)?

File: crates/oxc_linter/src/rules/eslint/func_names.rs [hunk 2]
```diff
@@ -142,6 +156,34 @@ fn is_object_or_class_method(parent_node: &AstNode) -> bool {
     }
 }
 
+// enum InferredName<'a> {
```
Comment (mysteryven) [inline crates/oxc_linter/src/rules/eslint/func_names.rs:159]: It looks like do the same as `guess_function_name`, is `InferedName` preparing for the future work?

Comment (DonIsaac) [reply crates/oxc_linter/src/rules/eslint/func_names.rs:159]: Oops forgot to remove that, thanks!

**Pull request merged**